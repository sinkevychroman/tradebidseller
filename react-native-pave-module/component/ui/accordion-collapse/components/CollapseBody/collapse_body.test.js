import 'react-native';
import React from 'react';

import CollapseBody from './index';
import renderer from 'react-test-renderer';
import {Text, View} from 'react-native';

describe('Collapse Body', () => {
  it('Should render CollapseBody', () => {
    renderer.create(<CollapseBody>Test</CollapseBody>);
  });
  it('Should match snapshot', () => {
    const renderedComponent = renderer.create(
      <CollapseBody>Test</CollapseBody>,
    );
    expect(renderedComponent.toJSON()).toMatchSnapshot();
  });
  it('Should render children correctly', () => {
    const renderedComponent = renderer.create(
      <CollapseBody>
        <Text>test</Text>
      </CollapseBody>,
    );
    expect(renderedComponent.root.props.children.type).toBe(Text);
    expect(renderedComponent.root.props.children.props.children).toBe('test');
  });
  it('Should have same type when wrapped inside a view', () => {
    const renderedComponent = renderer.create(
      <View>
        <CollapseBody>Test</CollapseBody>
      </View>,
    );
    expect(renderedComponent.root.props.children.type).toBe(CollapseBody);
  });
});
