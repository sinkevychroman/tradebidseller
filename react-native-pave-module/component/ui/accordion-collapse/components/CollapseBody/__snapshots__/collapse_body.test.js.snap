// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse Body Should match snapshot 1`] = `
<View>
  Test
</View>
`;
